src/main/kotlin/snyk/container/YAMLImageExtractor.kt

```diff
@@ -0,0 +1,66 @@
+package snyk.container
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiManager
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.util.io.readText
+import org.jetbrains.yaml.YAMLFileType
+import org.jetbrains.yaml.psi.YAMLFile
+import org.jetbrains.yaml.psi.YAMLKeyValue
+import java.lang.UnsupportedOperationException
+import java.util.stream.Collectors
+import kotlin.streams.toList
+
+object YAMLImageExtractor {
+
+    fun extractImages(file: YAMLFile): List<String> {
+        val list = mutableListOf<String>()
+        val children = PsiTreeUtil.findChildrenOfAnyType(file, YAMLKeyValue::class.java)
+
+        // apparently, sometimes the YAML plugin doesn't populate the document/elements correctly :(
+        // so we do some manual fallback parsing
+        if (children.isEmpty()) {
+            list.addAll(extractImagesFromText(file))
+        }
+
+        for (child in children) {
+            if (child.keyText == "image") {
+                list.add(child.valueText)
+            }
+        }
+
+        return list
+    }
+
+    private fun extractImagesFromText(file: YAMLFile) =
+        try {
+            file.virtualFile.toNioPath().readText().split("\n").stream()
+                .filter { s -> s.trim().startsWith("image:") }
+                .map { s ->
+                    s.trim()
+                        .split(":")
+                        .stream().filter { e -> e.trim() != "image" }
```

Comment:
could we have image named `image` ?

Reply:
I'm not sure that this is allowed as image is a keyword.